src/styles/transformers/rtl.js

```diff
@@ -0,0 +1,90 @@
+import assign from 'lodash.assign';
+import warning from 'warning';
+
+const reTranslate = /((^|\s)translate(3d|X)?\()(\-?[\d]+)/;
+const reSkew = /((^|\s)skew(x|y)?\()\s*(\-?[\d]+)(deg|rad|grad)(,\s*(\-?[\d]+)(deg|rad|grad))?/;
+
+/**
+ * This function ensures that `style` supports both ltr and rtl directions by
+ * checking `styleConstants` in `muiTheme` and replacing attribute keys if
+ * necessary.
+ */
+export default (muiTheme) => (style) => {
+  if (process.env.NODE_ENV !== 'production') {
+    warning(!style.didFlip, `You're calling ensureDirection() on the same style
+      object twice.`);
+
+    style = assign({
+      didFlip: 'true',
+    }, style);
+  }
+
+  // Left to right is the default. No need to flip anything.
+  if (!muiTheme || !muiTheme.isRtl) return style;
```

Comment:
@oliviertassinari I agree we don't need the line but also agree with @alitaheri  that we should possibly leave the muiTheme in the transformers for now.

The reason why the `muiTheme` was there in the first place is because this `ensureDirections` now names (rtl) was left untouched minus the name and merge implementation. I believe the rtl check could be removed and the warning show be moved top level too (these are currently listed as remaining task at the top of the PR)